quill-async/src/main/scala/io/getquill/sources/async/Encoders.scala

```diff
@@ -8,19 +8,30 @@ import org.joda.time.LocalDateTime
 trait Encoders {
   this: AsyncSource[_, _, _] =>
 
+  abstract class Encoder[T](val f: T => Any)
```

Comment:
It makes impossible to infer some encoder described [here](https://github.com/getquill/quill/pull/165#issuecomment-184572327)